modules/adblocker/sources/adblocker.es

```diff
@@ -99,6 +116,26 @@ export default class Adblocker {
     );
   }
 
+  installHandlers() {
+    this.removeHandlers();
+
+    if (USE_PUSH_INJECTIONS_ON_NAVIGATION_EVENTS) {
+      this._onCommittedHandler = (details) => {
+        if (this.shouldProcessOnCommittedEvent(details)) {
+          this.manager.engine.onCommittedHandler(browser, details);
+        }
+      };
+      browser.webNavigation.onCommitted.addListener(this._onCommittedHandler);
```

Comment:
the canonical way to handle `onCommitted` events in the `common` is to add `content:location-change` event handler in the module background - like here https://github.com/ghostery/common/blob/f1d4bcf465540c879185ba9c512291c11382bc53/modules/human-web/sources/background.es#L195

That method will remove the listener when needed automatically

Reply:
By managing it locally in adblocker.es, the lifetime is bound to whether adblocking is enabled (users can turn it off and on). Looks more natural to me.

Reply:
Just noticed that the lifecycle argument does not hold: ghostery-extension disables the adblocker module, so it will be the same in both cases and indeed it would avoid the cleanup code.

One aspect why I would still prefer to avoid any abstraction on top of the browser API is that latency is a concern here: for instance, if commons introduces async operations in the handling of onCommitted, it can be problematic. The code-path from the event being emitted by the browser to the adblocker engine's call to execute scripts should be synchronous only. With each extra layer in between, I find it harder to reason about that.